sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -250,6 +250,103 @@ joint_matrix_load(Group sg,
 #endif // defined(__SYCL_DEVICE_ONLY__)
 }
 
+template <typename Group, typename S, typename T, size_t NumRows,
+          size_t NumCols, typename PropertyListT,
+          std::enable_if_t<std::is_same<S, std::remove_const_t<T>>::value,
+                           bool> = true>
+inline __SYCL_ALWAYS_INLINE void joint_matrix_load(
+    Group sg,
+    joint_matrix<Group, S, use::accumulator, NumRows, NumCols,
+                 sycl::ext::oneapi::experimental::matrix::layout::dynamic> &res,
+    ext::oneapi::experimental::annotated_ptr<T, PropertyListT> src,
+    size_t stride, sycl::ext::oneapi::experimental::matrix::layout Layout) {
+#if defined(__SYCL_DEVICE_ONLY__)
+#if defined(__NVPTX__)
+  std::ignore = sg;
+  throw runtime_error("Use joint_matrix_load on multi_ptr on Nvidia device.",
+                      PI_ERROR_INVALID_DEVICE);
+#elif defined(__HIP_PLATFORM_AMD_MFMA__)
+  throw runtime_error("Use joint_matrix_load on multi_ptr on AMD device.",
+                      PI_ERROR_INVALID_DEVICE);
+#else
+  std::ignore = sg;
+  T *Ptr = src.get();
+  switch (Layout) {
+  default:
+    assert(false && "Invalid Memory Layout!");
+  case layout::row_major:
+    res.spvm = __spirv_JointMatrixLoadINTEL<
+        T, S, NumRows, NumCols, spv_matrix_use_traits<use::accumulator>::value,
+        spv_matrix_layout_traits<layout::dynamic>::value>(
+        Ptr, stride, __spv::MatrixLayout::RowMajor,
```

Comment:
I suggest using this struct to avoid switch/case:

```suggestion
        Ptr, stride, spv_matrix_layout_traits<Layout>::value,
```

Reply:
This cannot be used as layout is not constexpr

Reply:
Indeed, I missed that.

Reply:
We can define something like this then in detail namespace:
```
inline __SYCL_ALWAYS_INLINE __spv::MatrixLayout joint_matrix_layout_to_spv(
    sycl::ext::oneapi::experimental::matrix::layout Layout) {
  switch (Layout) {
  case layout::row_major:
    return __spv::MatrixLayout::RowMajor;
  case layout::col_major:
    return __spv::MatrixLayout::ColumnMajor;
  case layout::ext_intel_packed:
    return __spv::MatrixLayout::Packed;
  case layout::dynamic:
    return __spv::MatrixLayout::Dynamic;
  }
}
```
Then, instead of repeating switch-case and the same calls across files, we could use something like this:
```suggestion
        Ptr, stride, joint_matrix_layout_to_spv(Layout),
```